03_method/02_define.rb

```diff
@@ -11,8 +16,49 @@
 #   - 引数がnilの場合は、dev_teamメソッドを呼ぶこと
 # - また、2で定義するメソッドは以下を満たすものとする
 #   - メソッドが定義されるのは同時に生成されるオブジェクトのみで、別のA2インスタンスには（同じ値を含む配列を生成時に渡さない限り）定義されない
+class A2
+  def initialize(array)
+    # オブジェクトの特異クラスにメソッドを定義する
+    array.each do |val|
+      val_with_prefix = "hoge_#{val}"
+      define_singleton_method val_with_prefix.to_sym do |num|
+        return dev_team if num.nil?
+
+        val_with_prefix * num
```

Comment:
```suggestion
        __method__.to_s * num
```
`__method__`でメソッド名を取得することができる

https://docs.ruby-lang.org/ja/latest/method/Kernel/m/__method__.html